doc/build.md

```diff
@@ -54,6 +56,18 @@ $ cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_PREFIX_PATH="$(pwd)/../third-p
 $ cmake --build . -- -j $(nproc)
 ```
 
+### Mac build commands
+Mac environments default to a proprietary version of LLVM and Clang which do not work with ML-IO build. It is recommended to use the LLVM that is installed via Homebrew; this version is usually found in ```/usr/local/opt/llvm/bin/clang``` after installation.
+
+```bash
+$ git clone https://github.com/awslabs/ml-io.git mlio
+$ cd mlio
+$ build-tools/build-dependency build/third-party all
+$ mkdir build/release
+$ cd build/release
+$ cmake -DCMAKE_C_COMPILER=/usr/local/opt/llvm/bin/clang -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_PREFIX_PATH="$(pwd)/../third-party" ../..
```

Comment:
You should also add
```
-DCMAKE_CXX_COMPILER=/usr/local/opt/llvm/bin/clang++
```

Reply:
Why not using: sysctl -n hw.ncpu?

Reply:
When I was building, Cmake inferred the cxx compiler from the c compiler, because it correctly identified the cxx compiler without it. But I can retry to double check.

Reply:
I'll add it just to be safe.